application/common/models/User.php

```diff
@@ -298,6 +298,15 @@ private function validatePassword(): Closure
             $currentPassword = $this->currentPassword ?? new Password();
             if (! password_verify($this->password, $currentPassword->hash)) {
                 $this->addError($attributeName, 'Incorrect password.');
+            } else {
+                // check the current hash cost and rehash if necessary
+                if (password_needs_rehash(
+                    $currentPassword->hash,
+                    Password::HASH_ALGORITHM,
+                    ['cost' => Password::HASH_COST]
+                )) {
+                    $this->savePassword();
```

Comment:
Another interesting detail of this is that it will appear that the user changed their password anytime we do this, when in fact we simply re-hashed their existing password.

Is there a way would could instead change the hash on their existing password record? (Feel free to suggest alternatives.)